client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java

```diff
@@ -313,6 +315,109 @@ public void onFailure(Exception e) {
         assertTrue(latch.await(30L, TimeUnit.SECONDS));
     }
 
+    public void testExplainLifecycle() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        // create a policy & index
+        {
+            Map<String, Phase> phases = new HashMap<>();
+            Map<String, LifecycleAction> hotActions = new HashMap<>();
+            hotActions.put(RolloverAction.NAME, new RolloverAction(
+                new ByteSizeValue(50, ByteSizeUnit.GB), null, null));
+            phases.put("hot", new Phase("hot", TimeValue.ZERO, hotActions));
+
+            LifecyclePolicy policy = new LifecyclePolicy("my_policy",
+                phases);
+            PutLifecyclePolicyRequest putRequest =
+                new PutLifecyclePolicyRequest(policy);
+            client.indexLifecycle().putLifecyclePolicy(putRequest, RequestOptions.DEFAULT);
+
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest("my_index",
+                Settings.builder()
+                    .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put("index.lifecycle.name", "my_policy")
+                    .build());
+            client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+            CreateIndexRequest createOtherIndexRequest = new CreateIndexRequest("other_index",
+                Settings.builder()
+                    .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                    .build());
+            client.indices().create(createOtherIndexRequest, RequestOptions.DEFAULT);
+
+
+            // wait for the policy to become active
+            assertBusy(() -> assertNotNull(client.indexLifecycle()
+                .explainLifecycle(new ExplainLifecycleRequest("my_index"), RequestOptions.DEFAULT)
+                .getIndexResponses().get("my_index").getAction()));
+        }
+
+        // tag::ilm-explain-lifecycle-request
+        ExplainLifecycleRequest request =
+            new ExplainLifecycleRequest("my_index", "other_index"); // <1>
+        // end::ilm-explain-lifecycle-request
+
+        // tag::ilm-explain-lifecycle-execute
+        ExplainLifecycleResponse response = client.indexLifecycle()
+            .explainLifecycle(request, RequestOptions.DEFAULT);
+        // end::ilm-explain-lifecycle-execute
+        assertNotNull(response);
+
+        // tag::ilm-explain-lifecycle-response
+        Map<String, IndexLifecycleExplainResponse> indices =
```

Comment:
I'm not sure what would be different in terms of what we show here - it will just have different values (`isManaged = false`, everything else is null), which we don't show in the docs.  Do you just mean we should add `indices.get("other_index")` to show that you can do it?

Reply:
I was thinking of adding anchored comments discussing how things will be null or something. Maybe adding it in the description?

Reply:
Yeah, with adding it to the description makes more sense to me - I'll also add some asserts to that effect.

Reply:
👍